hnn_core/network.py

```diff
@@ -1170,6 +1196,74 @@ def __repr__(self):
 
         return entr
 
+    def drop(self, probability):
```

Comment:
honestly I would keep this for future PR. I think you can add `probability` to `add_connection` with 1 or 2 lines and expose it in an example. This falls under the usecase of modifying existing connections which we haven't started looking into yet.

Reply:
But we'll probably want to modify the "default" connections probabilistically, too?